### PR TITLE
more consistent interfaces

### PR DESCRIPTION
**Pull request by tiemvanderdeure**
Implement interface functions sdm_ensemble, evaluate, and predict, with basic documentation and better dispatches.

### CODE REVIEW

File: src/data_utils.jl [hunk 1]
```diff
@@ -0,0 +1,14 @@
+### Miscelanious utilities to deal with data issues such as names, missing values
+
+function _get_predictor_names(p, a)
+    predictors = Base.intersect(Tables.schema(a).names, Tables.schema(p).names)
+    predictors = filter!(!=(:geometry), predictors) # geometry is never a variable
+    length(predictors) > 0 || error("Presence and absence data have no common variable names - can't fit the ensemble.")
+    return predictors
+end
+
+function _missingmask(rs::Rasters.AbstractRasterStack)
```
Comment (rafaqz) [inline src/data_utils.jl]: Rasters.missingmask does this (maybr just on the first band? We could add a keyword fir that tho)

File: src/predict.jl [hunk 2]
```diff
@@ -1,21 +1,111 @@
...
+function _build_raster(missing_mask::Rasters.AbstractRaster, pr::Vector)
+    r_dims = Rasters.dims(missing_mask)
+    T = eltype(pr) # usually Float64, but could be something else depending on the reducer
+    output = Raster(Array{Union{Missing, T}}(missing, size(r_dims)), dims = r_dims)
+    output[missing_mask] .= pr
+    return output
+end
+
+
 #= on pause until RasterStacks are compatible with Tables.jl
```
Comment (rafaqz) [inline src/predict.jl]: They are already?

File: src/predict.jl [hunk 3]
```diff
@@ -1,21 +1,111 @@
...
-        _unsafe_predict(mach, data_)
+function _reformat_and_predict_raster(s::Union{<:SDMensemble, SDMgroup, SDMmachine}, rs::Rasters.AbstractRasterStack, args...)
+    missing_mask = _missingmask(rs)
+    d = rs[missing_mask]
+    pr =  _reformat_and_predict(s, d, args...)
+    return _build_raster(missing_mask, pr)
+end
+
+# Build Raster with the models/machines in Band dimension
+function _build_raster(missing_mask::Rasters.AbstractRaster, pr::NamedTuple)
```
Comment (rafaqz) [inline src/predict.jl:86]: Why raster bands instead of stack layers?

So its more easily saved to tif?

File: src/data_utils.jl (hunk 1 above)
Comment (tiemvanderdeure) [reply src/data_utils.jl]: Rasters.missingmask only works on the first band. But yes I agree that this should really be in Rasters

File: src/predict.jl (hunk 2 above)
Comment (tiemvanderdeure) [reply src/predict.jl]: just some old stuff I forgot to remove

File: src/predict.jl (hunk 3 above)
Comment (tiemvanderdeure) [reply src/predict.jl:86]: That's what I had in mind, yes. And also so it's trivial to apply functions across the Band dimension.

File: src/data_utils.jl (hunk 1 above)
Comment (rafaqz) [reply src/data_utils.jl]: Like `alllayers=true` ? Probably that should be the default

Comment (tiemvanderdeure) [reply src/data_utils.jl]: I can write a PR

Comment (tiemvanderdeure) [reply src/data_utils.jl]: Fixed now!

File: src/predict.jl (hunk 3 above)
Comment (rafaqz) [reply src/predict.jl:86]: Yeah we need to make broadcast work on stacks

**Pull request merged**